src/simulation/network/network_partition.rs

```diff
@@ -4,7 +4,7 @@ use crate::simulation::network::node::Node;
 use std::collections::{BTreeMap, HashSet};
 use std::fmt::Debug;
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Clone is not needed any more but I thought it should be no problem, if it also derives `Clone` so I didn't revert it.